src/components/ImportForm/SecretSection/SecretSection.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { useK8sWatchResource } from '@openshift/dynamic-plugin-sdk-utils';
+import { FormGroup, TextInputTypes, Flex, FlexItem, Button } from '@patternfly/react-core';
+import { PlusCircleIcon } from '@patternfly/react-icons/dist/js/icons/plus-circle-icon';
+import { useFormikContext } from 'formik';
+import { SPIAccessTokenBindingGroupVersionKind } from '../../../models';
+import { InputField, TextColumnField } from '../../../shared';
+import { SPIAccessTokenBindingKind } from '../../../types';
+import { useModalLauncher } from '../../modal/ModalProvider';
+import { SNYK_SPI_TOKEN_ACCESS_BINDING } from '../../Secrets/secret-utils';
+import { SecretModal } from '../../Secrets/SecretModal';
+import { ImportFormValues } from '../utils/types';
+
+const SecretSection = () => {
+  const showModal = useModalLauncher();
+  const { values, setFieldValue } = useFormikContext<ImportFormValues>();
+
+  const [secret] = useK8sWatchResource<SPIAccessTokenBindingKind>({
+    groupVersionKind: SPIAccessTokenBindingGroupVersionKind,
+    namespace: values.namespace,
+    name: SNYK_SPI_TOKEN_ACCESS_BINDING,
+  });
+  const partnerTaskSecrets = secret?.status?.syncedObjectRef?.name
+    ? [secret.status.syncedObjectRef.name]
+    : [];
+
+  const onSubmit = React.useCallback(
+    (secretValue) => {
+      const secrets = [...values.secrets, secretValue];
+      const secretNames = [...values.newSecrets, secretValue.secretName];
+      setFieldValue('secrets', secrets);
+      setFieldValue('newSecrets', secretNames);
+    },
+    [values, setFieldValue],
+  );
+
+  return (
+    <FormGroup>
+      <TextColumnField
+        name="newSecrets"
+        label="Secrets"
+        addLabel="Add secret"
+        placeholder="Secret"
+        helpText="Keep your data secure by defining a build time secret"
+        noFooter
+        onChange={(v) =>
+          setFieldValue(
+            'secrets',
+            values.secrets.filter((vs) => v.includes(vs.secretName)),
+          )
+        }
+      >
+        {(props) => <InputField name={props.name} type={TextInputTypes.text} isReadOnly />}
+      </TextColumnField>
+      <Flex direction={{ default: 'row' }} justifyContent={{ default: 'justifyContentFlexStart' }}>
```

Comment:
Do we have a way to show the already created secrets so that user doesn't create the same secrets again?

Reply:
I thought about this, but the problem is the list will be long if we show all the secrets (even if we filter by type - Opaque). some secrets are already created for the user and some are created by the users. It will show an alert message when you submit a secret with existing name and also I am checking if `snyk-secret` is already created or not, if its created I am not showing that secret in the predefined list.